File: src/name/soy/dc/utils/WSMsg.java
package name.soy.dc.utils;

import com.google.gson.JsonObject;
import org.jetbrains.annotations.NotNull;

import java.util.function.Supplier;

public enum WSMsg {
    INVALID_DEVICE(()->{
        JsonObject json = new JsonObject();
        json.addProperty("level","error");
        json.addProperty("data","设备不存在！");
        return json;
    }),
    DEVICE_OK(()->{
        JsonObject json = new JsonObject();
        json.addProperty("level","nice");
        return json;
    }),
    AIDA_DATA(()->{
        JsonObject json = new JsonObject();
        json.addProperty("level","nice");
        return json;
    }),
    INVALID_SESSION(()->{
        JsonObject json = new JsonObject();
        json.addProperty("level","error");
        json.addProperty("data","验证失败");
        return json;
    });
    private JsonObject json;




    WSMsg(@NotNull Supplier<JsonObject> o) {
        json = o.get();
        json.addProperty("type",name().toLowerCase());
    }

    public JsonObject get() {
        return json.deepCopy();
    }
}

File: src/name/soy/dc/utils/WSMsg.kt
package name.soy.dc.utils

import com.google.gson.JsonObject
import java.util.function.Supplier

enum class WSMsg(o: Supplier<JsonObject>) {
	INVALID_DEVICE(Supplier<JsonObject> {
		JsonObject().apply {
			addProperty("level", "error")
			addProperty("data", "设备不存在！")
		}
		
	}),
	DEVICE_OK(Supplier<JsonObject> {
		JsonObject().apply {
			addProperty("level", "nice")
		}
		
	}),
	AIDA_DATA(Supplier<JsonObject> {
		JsonObject().apply {
			addProperty("level", "nice")
		}
	}),
	INVALID_SESSION(Supplier<JsonObject> {
		val json = JsonObject()
		json.addProperty("level", "error")
		json.addProperty("data", "验证失败")
		json
	});
	
	private val json: JsonObject = o.get().apply {
		addProperty("type", name.toLowerCase())
	}
	operator fun invoke(): JsonObject {
		return json.deepCopy()
	}
}